com.unity.multiplayer.mlapi/Runtime/Core/NetworkTickSystem.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using UnityEngine;
+
+namespace MLAPI
+{
+    public class NetworkTickSystem : INetworkUpdateSystem, IDisposable
+    {
+        private const double k_DefaultTickDuration = 0.016;
+        private double m_TickDuration;
+        private int m_NetworkTick; //How many network ticks have passed?
+
+        private static NetworkTickSystem m_Instance = null;
+
+        // special value to indicate "No tick information"
+        public const ushort k_NoTick = ushort.MaxValue;
+        // Number of ticks over which the tick number wraps back to 0
+        public const ushort k_TickPeriod = k_NoTick - 1;
+
+        public static NetworkTickSystem Instance
+        {
+            get
+            {
+                if (m_Instance == null)
+                {
+                    m_Instance = new NetworkTickSystem();
+                }
+                return m_Instance;
+            }
+        }
```

Comment:
singleton on its own vs an instance attached to a specific `NetworkManager`? I prefer the latter, what do you think? :)

Reply:
Attached to a specific NetworkManager is cleaner.
But the singleton PR has not landed yet. So, I'd rather push this as-is, and when we have converged on the correct way to deal with singletons, we replicate the pattern for this, too? Workable?

Reply:
you can still put instances under `NetworkManager` and access via `NetworkManager.Singleton` later on right?

Reply:
Why not instantiate it within the NetworkUpdateLoop itself?  That was my original plan, have it accessible via NetworkUpdateLoop.  Really, this is not the final version for the tick system, so NetworkManager or NetworkUpdateLoop... one of the two would be nice.

Reply:
`NetworkUpdateLoop` will always be a static class but `NetworkManager` is not a static class and will not be a singleton in the future so I'd suggest putting this instance under `NetworkManager` instead of `NetworkUpdateLoop` — also I don't want to repeat singleton pattern, we should be trying to avoid and remove them, not introduce them IMHO.

Reply:
How about we go with this pattern for now but link in the Singleton removal ticket here in the comment so we remember to revise when that goes in

Reply:
See how the RPCQueueContainer is instantiated

Reply:
Addressed in latest update. Thanks!